Dockerfile.dapper

```diff
@@ -1,11 +1,11 @@
-FROM registry.suse.com/bci/golang:1.22.8
+FROM registry.suse.com/bci/golang:1.23
```

Comment:
Should this be pinned to 1.23.4?

Reply:
From other related PRs, seems bci base image is 1.23 now, it may be slightly different with go version itself

https://github.com/rancher/support-bundle-kit/pull/132/files#diff-3598233dce47ce9c207a77a9f23394d45235397dd66d7dd9de909487e1bd8e8aL1